### PR TITLE
fix(frontend): links instead of js events in admin

### PR DESCRIPTION
**Pull request by lpoujade**
Issue #3953

Replace the javascript events by html links in admin spaces list

I could not run the pre-commit scripts because of requirement of https://gitlab.com/pycqa/flake8.git/ which doesn't seems to exists

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link) => could be added but this is a little change, and as I'm not totally sure it is the wanted behavior I'll wait for your opinion
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done
 - See https://github.com/tracim/tracim/pull/6149#issuecomment-1521393369

### CODE REVIEW

**Comment by Haskaris** [comment]
Hello, thanks for taking the time to improve Tracim!

Indeed, looks like the repository have migrated from gitlab to [github](https://github.com/pycqa/flake8)
Updating the ``.pre-commit-config.yaml`` to replace ``https://gitlab.com/pycqa/flake8`` to ``https://github.com/pycqa/flake8``.

**Comment by lpoujade** [comment]
Done, it seems OK (ran with my changes staged):

```
[INFO] Stashing unstaged files to /home/low/.cache/pre-commit/patch1681890082-449694.
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check JSON...........................................(no files to check)Skipped
Check Yaml...............................................................Passed
Debug Statements (Python)............................(no files to check)Skipped
Check for merge conflicts................................................Passed
black................................................(no files to check)Skipped
seed isort known_third_party.............................................Passed
isort................................................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
```

File: frontend_app_admin_workspace_user/src/component/AdminWorkspace.jsx
Comment (Skylsmoi) [inline frontend_app_admin_workspace_user/src/component/AdminWorkspace.jsx:160]: You cannot use the `<a>` tag here because all redirections are made with ReactRouter.
You must use the `<Link>` component from react-router-dom here.

Also, I'd like you to add the classes `primaryColorFont` and `primaryColorFontDarkenHover` to style the text color please.
You'll can then also remove the `primaryColorFontHover` from the parent's `<td>`

File: frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx
Comment (Skylsmoi) [inline frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx:584]: since `workspaceIdOpened` isn't used anymore, you can remove it's initialization in the cronstructor

**Review by Skylsmoi — CHANGES_REQUESTED** [review]
Thank for your work.
The change requested for the `<Link>` instead of a `<a>` is mandatory though

**Comment by lpoujade** [comment]
Aah ok, I missed that ! I'll try to fix this by friday

Le 19 avril 2023 16:09:18 GMT+02:00, Skylsmoi ***@***.***> a écrit :

File: frontend_app_admin_workspace_user/src/component/AdminWorkspace.jsx
Comment (lpoujade) [reply frontend_app_admin_workspace_user/src/component/AdminWorkspace.jsx]: solved in [95b7c6a](https://github.com/tracim/tracim/pull/6149/commits/95b7c6a1d072be4bbd2675884c143f88cf7f3819)

File: frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx
Comment (lpoujade) [reply frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx:584]: solved in [95b7c6a](https://github.com/tracim/tracim/pull/6149/commits/95b7c6a1d072be4bbd2675884c143f88cf7f3819)

Comment (Skylsmoi) [reply frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx:584]: confirmed, you can resolve

**Review by Skylsmoi — APPROVED** [review]
I approve these changes. Good work.

I also have tested them on your branch.

This PR is ready to be merged.
We are in the process of a new release at this moment and we will try to include your fix in the upcoming release.

Thank you for your participation to Tracim!

**Comment by Haskaris** [comment]
Hello
After more testing, it appears that there is some regressions :confused: 

1. The color of the name space/link is always green - `primaryColor`. To resolve that you can do the opposite as told in https://github.com/tracim/tracim/pull/6149#discussion_r1171391874 and remove `primaryColorFont`.
This will however lead to a blue color (link color). To resolve this, you will have to set the color text of the `Link` to `inherit`
2. The clickable size of the link is shrink to the text size, while the whole cell is supposed to be clickable.
To resolve this, you can try:
- To play around the margin/padding size (you can take inspiration from `notification__list__item__circle` in `NotificationItem.jsx` and `NotificationWall.styl`) to increase the clickable size
- To update the Link size using classes to fit the `td` size (_probably the best approach here_)

**Comment by lpoujade** [comment]
Hello, OK I can fix this monday or tuesday :)

Le 25 avril 2023 10:33:46 GMT+02:00, Mathis Perrier ***@***.***> a écrit :

**Comment by lpoujade** [comment]
>     1. The color of the name space/link is always green - `primaryColor`. 
We want grey links, and green when hovered, right ?

**Comment by Skylsmoi** [comment]
@lpoujade yes

But to be even more accurate, we want "default font color" links, and primaryColorFontHover when hovered.

"green" comes from `.primaryColor` which is the default color that you can overrides in `/frontend/dist/assets/branding/color.json`.
The backend then generates a list of classes with darken and lighten variations to be used for background, border, font....

The class .primaryColorFontHover is the darkened variation of the primaryColor you should use for hovered text.

**Review by Millefeuille42 — APPROVED** [review]
Approved overall, previous conversations must be resolved though

**Comment by lpoujade** [comment]
I'm on it, I have a simple solution using margins but I'm trying to find a better one with a correct Link size

File: frontend_app_admin_workspace_user/src/css/index.styl
Comment (Skylsmoi) [inline frontend_app_admin_workspace_user/src/css/index.styl:39]: the alignment fix is not necessary, I agree it's better your way but it would require additional fix on other columns.

I'd suggest you to replace the 2 highlighted lines by these lines:
```css
display inline-block
width 100%
```
so that their won't be any regression from the previous code. After that I'll merge your PR.

**Review by Skylsmoi — CHANGES_REQUESTED** [review]

**Comment by PhilippeAccorsi** [comment]
I think this fix is a bugfix for tracim 4.5.x so maybe better to merge in maintenance branch.

**Comment by lpoujade** [comment]
I'll try this at the end of the week; I may be wrong but from memory `width 100%` didn't work because of the `max-width 0` on the parent

Le 10 mai 2023 16:25:22 GMT+02:00, Skylsmoi ***@***.***> a écrit :

**Comment by lpoujade** [comment]
I had half of the thing: it seems that using inline-block, the width seems ok, but the height doesn't expand and `text-overflow ellipsis` is broken (using Firefox 112.0.2/113.0b9 and Chromium 113.0.5672.63)

**Comment by Skylsmoi** [comment]
You are correct about the ellipsis, the height isn't an issue though.
Your solution works better, my solution was only to avoid "hacking" margins and padding with negative values.
Let's keep it your way, I'll handle the merging because, as Philippe said, we will have a maintenance branch for our most recent release, 4.5.0.

**Review by Skylsmoi — APPROVED** [review]

**Comment by lpoujade** [comment]
About the height, I meant that the clickable area of the link only takes the height of the text


Le 17 mai 2023 10:10:41 GMT+02:00, Skylsmoi ***@***.***> a écrit :

**Comment by lebouquetin** [comment]
@Skylsmoi is there a reason for not merging this? It looks like we did not merge it because of time conflict between new release about to be delivered and timing of the PR. Now my feeling is we can merge.

Do you confirm this?

**Comment by Skylsmoi** [comment]
I confirm
I think I forgot it

**Pull request merged**
